src/include/util/set_util.h

```diff
@@ -0,0 +1,46 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// set_util.h
+//
+// Identification: src/include/util/set_util.h
+//
+// Copyright (c) 2015-18, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <set>
+
+namespace peloton {
+
+class SetUtil {
+ public:
+  template <typename T>
+  static bool IsDisjoint(const std::set<T> &setA, const std::set<T> &setB);
+};
+
+template <typename T>
+inline bool SetUtil::IsDisjoint(const std::set<T> &setA,
+                                const std::set<T> &setB) {
+  bool disjoint = true;
+  if (setA.empty() || setB.empty()) return disjoint;
+
+  typename std::set<T>::const_iterator setA_it = setA.begin();
+  typename std::set<T>::const_iterator setB_it = setB.begin();
```

Comment:
You can use `auto` here to reduce visual noise.

Reply:
I thought it doesn't work in C++11 but it does!! great! :) I will fix it.